### PR TITLE
Add new sample for Vulkan Profiles Library

### PR DESCRIPTION
**Pull request by SaschaWillems**
## Description

This PR adds a sample for using the new Vulkan Profiles Library. The sample overrides instance and device creation (from the framework) to use the profile functions instead. The sample also includes a small tutorial.

Note: This required a few changes to the framework.

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly
- [x] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [x] I have tested the sample on at least one compliant Vulkan implementation
- [x] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [x] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [x] Any dependent assets have been merged and published in downstream modules
- [x] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

### CODE REVIEW

**Review by gary-sweet — COMMENTED** [review]
Correctly reports that we don't support the entire profile, but that's as much as I can test right now.

**Review by gpx1000 — DISMISSED** [review]

**Comment by marty-johnson59** [comment]
@aurolou, @mklefrancois, could you guys take a look at this sample and approve if OK?  Right now, it only works on DT systems, so could use your help to get this one approved/released.  Thanks!

File: samples/tooling/profiles/README.md
Comment (JoseEmilio-ARM) [inline samples/tooling/profiles/README.md]: `vkCreateInstance`?

File: framework/core/device.h
Comment (JoseEmilio-ARM) [inline framework/core/device.h]: if possible, clean-up these todo's?

File: framework/vulkan_sample.cpp
Comment (JoseEmilio-ARM) [inline framework/vulkan_sample.cpp:129]: maybe it would be cleaner to move this to the default implementation of `create_instance`?

File: framework/vulkan_sample.cpp
Comment (JoseEmilio-ARM) [inline framework/vulkan_sample.cpp]: similar here, happy with the override just need to clean-up the todo and maybe move the default call to `create_device`

File: framework/vulkan_sample.h
Comment (JoseEmilio-ARM) [inline framework/vulkan_sample.h]: missing doc

File: samples/tooling/profiles/profiles.cpp
Comment (JoseEmilio-ARM) [inline samples/tooling/profiles/profiles.cpp]: wondering if we should move these to the top of the file, as they are the main focus of the sample?

File: samples/tooling/profiles/profiles.cpp
Comment (JoseEmilio-ARM) [inline samples/tooling/profiles/profiles.cpp]: for my benefit, why are these new helpers needed now? do they duplicate framework code?

**Comment by mklefrancois** [comment]
Tested and works as expected. The overrides `create_instance()` and `create_device()` are simple to follow.

File: samples/tooling/profiles/profiles.cpp
Comment (mklefrancois) [inline samples/tooling/profiles/profiles.cpp:213]: Not important, but why not using VMA ?

**Review by mklefrancois — DISMISSED** [review]

**Review by aurolou — DISMISSED** [review]

**Review by JoseEmilio-ARM — CHANGES_REQUESTED** [review]
Great sample! please see comments above

File: samples/tooling/profiles/profiles.cpp
Comment (SaschaWillems) [reply samples/tooling/profiles/profiles.cpp]: No duplication. The code in those helpers was part of the base class device setup and not accessible from the outside. As I didn't want to duplicate that whole code in the sample, I moved the code into those helpers and use them both in the framework device class as well as the sample.

File: samples/tooling/profiles/profiles.cpp
Comment (SaschaWillems) [reply samples/tooling/profiles/profiles.cpp]: That's a good point. I'll move both to the top of the sample, right below the constructor/destructor.

File: samples/tooling/profiles/profiles.cpp
Comment (SaschaWillems) [reply samples/tooling/profiles/profiles.cpp:213]: Good question. I guess I just didn't think about using VMA. I may take a look at this after the sample is merged.

File: framework/vulkan_sample.cpp
Comment (SaschaWillems) [reply framework/vulkan_sample.cpp:129]: Not sure if that's possible. The default implementation is just an empty function that can be overridden by a sample.

**Pull request merged**
